File: VB/CodeBehind/XPO/InfiniteAsyncSource/Issues/DemoDataHelper.vb
Imports DevExpress.Xpo
Imports System
Imports System.Linq

Namespace Issues
    Public Module DemoDataHelper
        Public Sub Seed()
            Using uow = New DevExpress.Xpo.UnitOfWork()
                Dim users = OutlookDataGenerator.Users.[Select](Function(x)
                                                                    Dim split = x.Split(" "c)
                                                                    Return New User(uow) With {
                                                                        .FirstName = split(0),
                                                                        .LastName = split(1)
                                                                    }
                                                                End Function).ToArray()
                uow.CommitChanges()
                Dim rnd = New Random(0)
                Dim issues = Enumerable.Range(0, 1000).[Select](Function(i) New Issue(uow) With {
                    .Subject = OutlookDataGenerator.GetSubject(),
                    .UserId = users(rnd.Next(users.Length)).Oid,
                    .Created = Date.Today.AddDays(-rnd.Next(30)),
                    .Priority = OutlookDataGenerator.GetPriority(),
                    .Votes = rnd.Next(100)
                }).ToArray()
                uow.CommitChanges()
            End Using
        End Sub
    End Module
End Namespace

File: CS/CodeBehind/XPO/ServerMode/Issues/DemoDataHelper.cs
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;

namespace XPOIssues.Issues {
    public static class DemoDataHelper {
        public static void Seed() {
            using(var uow = new DevExpress.Xpo.UnitOfWork()) {
                var users = OutlookDataGenerator.Users
                    .Select(x => {
                        var split = x.Split(' ');
                        return new User(uow) {
                            FirstName = split[0],
                            LastName = split[1]
                        };
                    })
                    .ToArray();
                uow.CommitChanges();
                var rnd = new Random(0);
                var issues = Enumerable.Range(0, 1000)
                    .Select(i => new Issue(uow) {
                        Subject = OutlookDataGenerator.GetSubject(),
                        UserId = users[rnd.Next(users.Length)].Oid,
                        Created = DateTime.Today.AddDays(-rnd.Next(30)),
                        Priority = OutlookDataGenerator.GetPriority(),
                        Votes = rnd.Next(100),
                    })
                    .ToArray();
                uow.CommitChanges();
            }
        }
    }
}
